src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -461,4 +461,92 @@ describe("provider", () => {
       assert.notStrictEqual(hash, "");
     });
   });
+
+  describe("disconnect()", () => {
+    let provider: EthereumProvider;
+
+    [true, false].forEach(asyncRequestProcessing => {
+      describe(`asyncRequestProcessing: ${asyncRequestProcessing}`, () => {
+        beforeEach("Instantiate provider", async () => {
+          provider = await getProvider({
+            chain: { asyncRequestProcessing }
+          });
+        });
+
+        it("rejects requests after disconnect() is called", async () => {
+          provider.disconnect();
```

Comment:
Should we `await` `disconnect` here?

Without `await`ing it seems that we are guaranteeing that `disconnect` immediately and _synchronously_ stops accepting new requests. I don't _think_ that's what we really want to test here (I think we _should_ `await`).

Reply:
That's _exactly_ what we are testing.

Because the disconnect logic is synchronous up until we start to shutdown the blockchain, as I understand, the `provider.request` in this test should be executed in the same event as the synchronous portion of `disconnect()` (the call to `coordinator.stop()`). Happy to learn more if you disagree.

Maybe the test should have a better name and / or comment to explain this? Keen to hear your thoughts?

Reply:
The new comment clears it up. My suggestion here was really that I don't think we need to guarantee this sort of synchronous behavior. I'm okay with the guarantee, just not _convinced_ we need to go that far. Feel free to Resolve this comment if you are okay with it too! (as I suspect you are)

Reply:
Maybe we don't need to guarantee it, but if we change the behaviour, we probably want to look deep into what the implications are - don't want to leak internal errors (such as the db is not connected).